tests/storage/__init__.py

```diff
@@ -1,98 +1,88 @@
 import os
 
 import pytest
+from ert_shared import ERT_STORAGE
 from ert_shared.storage.blob_api import BlobApi
 from ert_shared.storage.model import Blobs, Entities
 from ert_shared.storage.rdb_api import RdbApi
+from ert_shared.storage.storage_api import StorageApi
 from sqlalchemy import create_engine
 from sqlalchemy.orm import Session
 
 
-@pytest.fixture(scope="session")
-def engine():
-    engine = create_engine("sqlite:///:memory:", echo=False)
-    engine.execute("pragma foreign_keys=on")
-    return engine
-
+@pytest.fixture
+def initialize_databases(tmpdir):
+    rdb_url = f"sqlite:///{tmpdir}/entities.db"
+    blob_url = f"sqlite:///{tmpdir}/blobs.db"
 
-@pytest.fixture(scope="session")
-def tables(engine):
-    Entities.metadata.create_all(engine)
-    Blobs.metadata.create_all(engine)
-    yield
-    Entities.metadata.drop_all(engine)
-    Blobs.metadata.drop_all(engine)
+    ERT_STORAGE.initialize(rdb_url=rdb_url, blob_url=blob_url)
 
 
 @pytest.fixture
-def db_connection(engine, tables):
-    """Returns an sqlalchemy session, and after the test tears down everything properly."""
-    connection = engine.connect()
-    transaction = connection.begin()
-
-    yield connection
+def apis(initialize_databases):
+    rdb_session = ERT_STORAGE.RdbSession()
+    blob_session = ERT_STORAGE.BlobSession()
 
-    transaction.rollback()
-    connection.close()
+    rdb_api = RdbApi(session=rdb_session)
+    blob_api = BlobApi(session=blob_session)
 
+    try:
+        yield (rdb_api, blob_api)
 
-@pytest.fixture(scope="module")
-def db_info(tmp_path_factory):
-    def add_blob(data):
-        ret = blob.add_blob(data)
-        blob.flush()
-        return ret.id
+        rdb_session.commit()
+        blob_session.commit()
+    except:
+        rdb_session.rollback()
+        blob_session.rollback()
+    finally:
+        rdb_session.close()
+        blob_session.close()
 
-    tmp_path = tmp_path_factory.mktemp("database")
 
-    db_lookup = {}
-    db_name = "test.db"
-    db_url = f"sqlite:///{tmp_path}/{db_name}"
+@pytest.fixture
```

Comment:
This should probably be a `@pytest.fixture(scope="session")` so we don't recreate the same database over and over. Then a `@pytest.fixture` will use the session-wide fixture to provide `{Blob,Rdb}Session` and rollback when we're done.

Reply:
Yes - maybe that would be nice.... But some of the tests are adding data - and removing data maybe.... then we still need to repopulate the data after each test. But yes - that is less costly than recreating the database! i will look into it 👍

Reply:
The changes made by tests shouldn't propagate to other tests, so it's a bug if it does that currently.

Reply:
it does not - but it was used an in-mem database spawning new for every request... but it was cumbersome to use due to only one in-mem database alive at the same time... It did not fly perfectly with our setup of using two databases and caused me to write code with it's only purpose in life supporting testing - and that i didn't like. I will try to fix 👍